### PR TITLE
#13 adds author association and assigns author when creating a post

### PR DESCRIPTION
**Pull request by elbartostrikesagain**
We want to associate an author to their posts and to their comments. A Post and Comment can each only have 1 Author. This would be the model and database setup.
- Validate a `Post` and `Comment` have an `Author`.
- Set the `current_user` as the author on a post when its created. (Note that comments can't be created yet via a controller so no need to handle that as part of this).

### CODE REVIEW

File: rails/db/schema.rb [hunk 1]
```diff
@@ -27,6 +28,8 @@
     t.text "content"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.integer "author_id"
+    t.string "admin_approved"
```
Comment (TheDread-Pirate-Roberts) [inline rails/db/schema.rb:32]: Where did this line come from?

File: rails/spec/models/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+RSpec.describe Comment, type: :model do
+  describe "author" do
+    let(:comment) { FactoryBot.create(:comment) }
+
+    it "validate presence" do
+      expect(comment).to be_valid
+      comment.update_column(:author_id, nil)
+      expect(comment).to be_invalid
+    end
```
Comment (TheDread-Pirate-Roberts) [inline rails/spec/models/comment_spec.rb:11]: We have a sad path test for when the author_id is nil, could we perhaps add a happy path test for when it's not nil?

Comment (TheDread-Pirate-Roberts) [reply rails/spec/models/comment_spec.rb:11]: ahhh yeah scratch that :)

**Comment by TheDread-Pirate-Roberts** [comment]
What should the relationship between users and authors be if a post and comment require both? Is there a context where a user would perform neither action?

File: rails/app/models/post.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
 class Post < ApplicationRecord
   has_many :comments
+  belongs_to :author, class_name: "User", foreign_key: "author_id"
```
Comment (TheDread-Pirate-Roberts) [inline rails/app/models/post.rb:3]: Why did you add the class name again?

**Comment by TheDread-Pirate-Roberts** [comment]
In regard to this comment I left https://github.com/TransviewLogistics/interviews-Zach_Alewel/pull/2#issuecomment-1515277609

and this is a super nit comment - I would probably not introduce the concept of an author if there is no difference between an author and user in terms of functionality. Just my 0.02. cents 🤷🏻‍♂️

File: rails/spec/factories/comment.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :comment do
+    content { "MyText" }
+    author { FactoryBot.create(:user) }
+    post
```
Comment (TheDread-Pirate-Roberts) [inline rails/spec/factories/comment.rb:5]: can you explain what this method call is doing? and do we need to include it in the post factory?

File: rails/spec/factories/posts.rb [hunk 5]
```diff
@@ -2,5 +2,6 @@
   factory :post do
     title { "MyString" }
     content { "MyText" }
+    author { FactoryBot.create(:user) }
```
Comment (TheDread-Pirate-Roberts) [inline rails/spec/factories/posts.rb:5]: I think we're going to need to call `post` here.

Comment (TheDread-Pirate-Roberts) [reply rails/spec/factories/posts.rb:5]: We should probably have some comments in the post factory since a post has_many comments.
